screens/UserScreens/ProductionDashboard.jsx

```diff
@@ -0,0 +1,111 @@
+import { firebase_db } from "../../firebase.config.js";
+
+export default function ProductionDashboardScreen({ route }) {
+    const [production, setProduction] = useState({});
+    const [admins, setAdmins] = useState([]);
+    const [participants, setParticipants] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const playCode = route.params.playCode;
+    const db = firebase_db;
+
+    useEffect(
+        () =>
+            onValue(
+                ref(db, `/productions/${playCode}`),
+                (snapshot) => {
+                    if (!snapshot.exists()) {
+                        setLoading(false);
+                        return;
+                    }
+                    const data = snapshot.val();
+                    setProduction(data);
+                    setLoading(false);
+                },
+                { onlyOnce: true }
+            ),
+        []
+    );
+
+    useEffect(() => {
+        onValue(ref(db, `/productions/${playCode}/admins`), async (snapshot) => {
+            if (!snapshot.exists()) return;
+            const adminData = snapshot.val();
+            const newAdmins = await Promise.all(
+                Object.keys(adminData).map(async (admin) => {
+                    return get(ref(db, `users/${admin}`))
+                        .then((userSnapshot) => {
+                            if (!userSnapshot.exists()) return;
+                            const userData = userSnapshot.val();
+                            return [userData.firstName, userData.lastName];
+                        })
+                        .catch((error) => {
+                            console.log("uh oh: ", error.message);
+                        });
+                })
+            );
+
+            setAdmins(newAdmins);
+        });
+
+        onValue(ref(db, `/productions/${playCode}/participants`), async (snapshot) => {
+            if (!snapshot.exists()) return;
+            const participantData = snapshot.val();
+            const newParticipants = await Promise.all(
+                Object.keys(participantData).map(async (participant) => {
+                    return get(ref(db, `users/${participant}`))
+                        .then((userSnapshot) => {
+                            if (!userSnapshot.exists()) return;
+                            const userData = userSnapshot.val();
+                            return [userData.firstName, userData.lastName];
+                        })
+                        .catch((error) => {
+                            console.log("uh oh: ", error.message);
+                        });
+                })
+            );
+
+            setParticipants(newParticipants);
+        });
+    }, [production]);
+
+    return (
+        <View className="flex-col">
+            <Text className="self-center text-3xl font-extrabold">Production Dashboard</Text>
+            {loading ? (
+                <ActivityIndicator color="#000000" size="large" />
```

Comment:
Plz use custom LoadingWheel Component unless there's a reason to not :)

Reply:
There is a very good reason for that. it is in your PR so I don't have access to it yet lol

Reply:
Good point lol